src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -217,6 +217,119 @@ .maxstack 8
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
 
+        [Fact]
+        public void TestPropertyNotAssignedInStructConstructor()
+        {
+            var comp = CreateCompilation(@"
+var x = new C();
+x.P = 10;
+x = new C();
+
+public struct C
+{
+    public C()
+    {
+        System.Console.WriteLine(""In C..ctor: "" + P);
+    }
+
+    public int P { get => field; set => field = value; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics(
+                // (10,51): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields.
+                //         System.Console.WriteLine("In C..ctor: " + P);
+                Diagnostic(ErrorCode.WRN_UseDefViolationThisSupportedVersion, "P").WithArguments("this").WithLocation(10, 51)
+                );
+            CompileAndVerify(comp, expectedOutput: @"In C..ctor: 0
+In C..ctor: 0").VerifyIL("C..ctor", @"
+{
+    // Code size       37 (0x25)
+    .maxstack  2
+    .locals init (int V_0)
+    IL_0000:  ldarg.0
+    IL_0001:  ldc.i4.0
+    IL_0002:  stfld      ""int C.<P>k__BackingField""
+    IL_0007:  ldstr      ""In C..ctor: ""
+    IL_000c:  ldarg.0
+    IL_000d:  call       ""int C.P.get""
+    IL_0012:  stloc.0
+    IL_0013:  ldloca.s   V_0
+    IL_0015:  call       ""string int.ToString()""
+    IL_001a:  call       ""string string.Concat(string, string)""
+    IL_001f:  call       ""void System.Console.WriteLine(string)""
+    IL_0024:  ret
+}
+");
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void TestPropertyIsAssignedInStructConstructor()
+        {
+            var comp = CreateCompilation(@"
+var x = new C();
+x.P = 10;
+x = new C();
+
+public struct C
+{
+    public C()
+    {
+        System.Console.WriteLine(""In C..ctor before assignment: "" + P);
```

Comment:
>System.Console.WriteLine [](http://example.com/codeflow?start=8&length=24)

Is doing this in the constructor and doing this twice helps testing what we want? It feels like the following scenario is just what we need:
``` 
var x = new C();
System.Console.WriteLine(x.P);

public struct C
{
    public C()
    {
        P = 5;
    }

    public int P { get => field; set => field = value; }
}
```

Reply:
I was trying to test the implicit initialization of the backing field. It's tested in IL but was observing the runtime behavior as well.

In this test I'm calling the struct constructor twice. if the implicit initialization didn't exist, the second constructor call will have `P = 10` for the first WriteLine.

Let me know if the IL verification is enough and I'll simplify the test.

Reply:
It looks like ```TestPropertyNotAssignedInStructConstructor``` verifies implicit initialization. I guess, testing the combination is useful, but testing suggested scenario is useful as well.